sdk/src/wallet/wallet/operations/address_generation.rs

```diff
@@ -124,16 +124,7 @@ impl Wallet {
                 .address
                 .hrp
                 .clone(),
-            None => {
-                self.client_options
-                    .read()
-                    .await
-                    .clone()
-                    .finish()
-                    .await?
-                    .get_bech32_hrp()
-                    .await?
-            }
+            None => self.client().get_bech32_hrp().await?,
```

Comment:
😍

Reply:
Wait, was there a reason it was like that? Why wasn't it using the client in the first place?

Reply:
I guess because this is a Wallet fn not an Account fn.

Reply:
Right

Reply:
Could have used an account client